exercises/pythagorean-triplet/example.rs

```diff
@@ -1,11 +1,19 @@
-pub fn find() -> Option<u32> {
-    for a in 1..1000 {
-        for b in (a + 1)..(1000 - a) {
-            let c = 1000 - (a + b);
+pub fn find(sum: u32) -> Option<Vec<[u32; 3]>> {
```

Comment:
I strongly believe Option is not necessary. If it's empty, empty vector is a perfectly reasonable return value and is not an error (such that we might use Option for)

Reply:
If this suggestion to remove Option is accepted, don't forget to also remove `option` from `pythagorean-triplets` config.json

Reply:
I was under the impression that using empty Vec falls under 'assigning special meaning to the basic types' (like in this [comment](https://github.com/exercism/rust/pull/548#issuecomment-398094945)).

Besides, isn't Option used to convey the absence of a value, not only for errors? Would be strange not to make use of a special type in this case.

Also, Vec could be changed for some other type in the future (like HashSet or some custom struct), but the Option API would remain

Reply:
> I was under the impression that using empty Vec falls under 'assigning special meaning to the basic types'

I'm not under this impression. The empty vector is a valid value that inhabits the `Vec<[u32; 3]>` type, just as any other value of that type does. It is not special in any way.

> Besides, isn't Option used to convey the absence of a value

Sure.

> Would be strange not to make use of a special type in this case.

It is strange to not be using a valid value of the type `Vec<[u32; 3]>`. Under no circumstance will this code ever return `Some(vec![])`, which means using an `Option` is a waste.

> Also, Vec could be changed for some other type in the future (like HashSet or some custom struct), but the Option API would remain

True, but we are not talking about the future, we are talking about this PR as it is proposed. Since `Vec<_>` is inhabitable by an empty vector, we should make use of that fact.

Reply:
> It is not special in any way.

Oops, forgot to mention why that makes it different from `""`. In `two-fer`, we assigned special meaning to `""` by saying that `""` must be special-cased by the implemented function (the user of the passed-in argument) in such a way that `""` is equivalent to `"you"` rather than itself.

In this exercise, one who wishes to examine the empty vector need not do any special casing. Operations such as `len` and iterating over it all act as they should. Therefore, no special meaning is needed and none should be assigned.

Reply:
Got it. Thanks for the explanation